common/src/main/java/org/apache/uniffle/common/storage/StorageTypePOJO.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.uniffle.common.storage;
+
+import org.apache.uniffle.proto.RssProtos.StorageInfo;
+
+/**
+ * The POJO stands for plain old java object. The suffix exists to avoid conflict with another StorageType class.
+ */
+public enum StorageTypePOJO {
```

Comment:
For other classes, is there similar problems?

Reply:
POJO is the concept of `Java bean`. I don't understand why we introduce this concept.

Reply:
Could we have an another name?

Reply:
`storageMedia` ?

Reply:
> > POJO is the concept of Java bean. I don't understand why we introduce this concept.
> 
> POJO is a not Java Bean, although it's always compared to Java Bean. The main reason to add suffix here is to avoid name conflicting, and the wrapper class for pb message falls into the POJO category.

Why do we have the name conflict?

Reply:
> StorageType
> 
> > > > POJO is the concept of Java bean. I don't understand why we introduce this concept.
> > > 
> > > 
> > > POJO is a not Java Bean, although it's always compared to Java Bean. The main reason to add suffix here is to avoid name conflicting, and the wrapper class for pb message falls into the POJO category.
> > 
> > 
> > Why do we have the name conflict?
> 
> StorageType is also defined in `org.apache.uniffle.storage.util`, which conflicts this class name.

POJO reminds me that this may be a java bean, it's weird for me. If they have diferent package names, there isn't conflict names.

Reply:
> > If they have diferent package names, there isn't conflict names.
> 
> Unlike scala, java doesn't support class name renaming in the import path, and you cannot import the same class name in one file even if they are from different packages. It's extremely strange to use a fully qualified class name in java file and that always indicates some code smell. And FYI, I have renamed all the `StorageTypePOJO` to `StorageMedia` due to @zuston's suggestion, which I think it's a better name.
> 
> P.S. you cannot do the following in one java file
> 
> ```
> import org.apache.uniffle.common.storage.StorageType
> import org.apache.uniffle.storage.util.StorageType
> ```

but you don't need import any of them.
You can use org.apache.uniffle.common.storage.StorageType and org.apache.uniffle.storage.util.StorageType directly.

Reply:
`StorageMedia` is ok for me.